test/e2e_node/probe_stress_test.go

```diff
@@ -0,0 +1,187 @@
+	framework.ExpectNoError(err, "Failed to start pod")
+
+	ginkgo.By("Verifying no containers restarted")
+	gomega.Consistently(ctx, func(ctx context.Context) error {
+		updatedPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		for _, containerStatus := range updatedPod.Status.ContainerStatuses {
+			if containerStatus.RestartCount > 0 {
+				return fmt.Errorf("container %s restarted %d times", containerStatus.Name, containerStatus.RestartCount)
+			}
+		}
+		return nil
+	}, probeStressWaitTime, 1*time.Second).Should(gomega.Succeed())
+
+	ginkgo.By("Test passed: no unexpected container restarts")
+}
+
+func createPodWithHTTPProbes(numContainers int) *v1.Pod {
+	return createProbeStressPod(numContainers, func(i int) (v1.Probe, []v1.ContainerPort, []string) {
+		port := int32(8080 + i)
+		probe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: "/",
+					Port: intstr.FromInt(int(port)),
+				},
+			},
+			PeriodSeconds:    probeStressPeriodSeconds,
+			TimeoutSeconds:   1,
+			SuccessThreshold: 1,
+			FailureThreshold: 3,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: port, Protocol: v1.ProtocolTCP}}
+		args := []string{"netexec", fmt.Sprintf("--http-port=%d", port)}
+		return probe, ports, args
+	})
+}
+
+func createPodWithTCPProbes(numContainers int) *v1.Pod {
+	return createProbeStressPod(numContainers, func(i int) (v1.Probe, []v1.ContainerPort, []string) {
+		port := int32(8080 + i)
+		probe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				TCPSocket: &v1.TCPSocketAction{
+					Port: intstr.FromInt(int(port)),
+				},
+			},
+			PeriodSeconds:    probeStressPeriodSeconds,
+			TimeoutSeconds:   1,
+			SuccessThreshold: 1,
+			FailureThreshold: 3,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: port, Protocol: v1.ProtocolTCP}}
+		args := []string{"netexec", fmt.Sprintf("--http-port=%d", port)}
+		return probe, ports, args
+	})
+}
+
+func createPodWithGRPCProbes(numContainers int) *v1.Pod {
+	return createProbeStressPod(numContainers, func(i int) (v1.Probe, []v1.ContainerPort, []string) {
+		port := int32(5000 + i)
+		probe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port: port,
+				},
+			},
+			PeriodSeconds:    probeStressPeriodSeconds,
+			TimeoutSeconds:   1,
+			SuccessThreshold: 1,
+			FailureThreshold: 3,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: port, Protocol: v1.ProtocolTCP}}
+		args := []string{"grpc-health-checking", fmt.Sprintf("--port=%d", port)}
+		return probe, ports, args
+	})
+}
+
+func createProbeStressPod(numContainers int, generator func(i int) (v1.Probe, []v1.ContainerPort, []string)) *v1.Pod {
+	podName := "probe-stress-" + string(uuid.NewUUID())
+	containers := make([]v1.Container, numContainers)
+
+	for i := 0; i < numContainers; i++ {
+		probe, ports, args := generator(i)
```

Comment:
What's the purpose of `generator` here? Would it work if `createProbeStressPod` take 3 more args instead?